### PR TITLE
Developer book

### PR DESCRIPTION
**Pull request by dantleech**
I think we should add a developer guide here. This would be a separate book, so there would be the User section (which I believe is what this currently is) and the Developer section.

We need this because developer documentation should be a first-class citizen and the quality of such a document should be important otherwise it will not be used.

For example I want to write a guide on how to write behat tests, standardize UI workflows (e.g. there should always be a success / error notification, delete buttons should always be here, etc.).

Doing this in markdown in the existing `/docs` repository which is of variable quality does not really appeal to me and browsing such documents on GITHub isn't the best experience - RST provides a much richer medium for technical documentation and readthedocs provides a great reading experience.

### CODE REVIEW

File: developer/behat/getting-started.rst [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    You will need to install the Java Runtime Environment. On a debian based
+    system you could do ``apt-get install default-jre``
+
+Selenium2 (headless, with PhantomJs)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+
+    It is not currently recommended to run tests on PhantomJS - the results
+    maybe unpredictable.
```
Comment (danrot) [inline developer/behat/getting-started.rst]: shouldn't it be `may be`?

File: developer/behat/getting-started.rst [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            # ...
+            Behat\MinkExtension:
+                sessions:
+                    default:
+                        selenium2:
+                            wd_host: "http://localhost:8643/wd/hub"
+
+.. note::
+
+    Tests run with phantom JS may be unstable.
```
Comment (danrot) [inline developer/behat/getting-started.rst]: duplicate note of above?

File: developer/behat/getting-started.rst [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+Running the tests
+-----------------
+
+You can run all the tests as follows:
+
+.. code-block:: bash
+
+    $ ./vendor/behat/behat/bin/behat -p<profile>
+
+Where ``<profile`` is one of:
```
Comment (danrot) [inline developer/behat/getting-started.rst]: `<profile>`

File: developer/behat/getting-started.rst (hunk 1 above)
Comment (dantleech) [inline developer/behat/getting-started.rst]: Yes

**Comment by dantleech** [comment]
Updated.

**Pull request merged**